KaitaiStream.cs

```diff
@@ -7,67 +7,92 @@
 namespace Kaitai
 {
     /// <summary>
-    /// The base Kaitai steam which exposes an API for the Kaitai Struct framework.
-    /// It's based off a <code>BinaryReader</code>, which is a little-endian reader.
+    /// The base Kaitai stream which exposes an API for the Kaitai Struct framework.
+    /// It's based off a <c>BinaryReader</c>, which is a little-endian reader.
     /// </summary>
     public partial class KaitaiStream : BinaryReader
     {
         #region Constructors
 
+        /// <summary>
+        /// Create a KaitaiStream backed by abstract stream. It could be in-memory buffer or open file.
+        /// </summary>
         public KaitaiStream(Stream stream) : base(stream)
         {
         }
 
-        ///<summary>
-        /// Creates a KaitaiStream backed by a file (RO)
-        ///</summary>
-        public KaitaiStream(string file) : base(File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read))
+        /// <summary>
+        /// Create a KaitaiStream backed by a currently closed file (in read-only binary-mode).
+        /// </summary>
```

Comment:
This "currently closed" is still very confusing. I've definitely not a Python person, but for C# guys that would be just as confusing. From OS point of view, "currently closed" file just does not exist as a resource: it takes no memory buffers, no handles, no nothing. Why don't we rephrase it in a somewhat more straightforward voice, i.e. something

> Create a KaitaiStream by opening a file in read-only binary mode. This KaitaiStream will be backed by FileStream.

Reply:
Okie dokie. Consider it fixed.

Reply:
Fixed.